### DISCUSSION TITLE
Introducing udata 4

### DISCUSSION THREAD

**Discussion by maudetes** [opendatateam/udata#2724] [Announcements]
See our blogpost about introducing udata 4: https://www.data.gouv.fr/fr/pages/udata/4/introducing-udata-4. :tada: 

If you are interested in udata 4, either to discuss architectural changes or to get started, you're at the right place ! :wave: 
Please comment below with an introduction of yourself and tell us about your interest in udata 4.

**Comment by Guts** [comment]
> See our blogpost about introducing udata 4: https://www.data.gouv.fr/fr/pages/udata/4/introducing-udata-4.md. tada

Hyperlink leads to a 404 page.

![image](https://user-images.githubusercontent.com/1596222/162237575-ea74f952-c15b-4379-b822-32df6acf1c29.png)

**Reply by maudetes** [reply]
Fixed, thank you for the notice!

Don't hesitate to comment on this discussion or tell us how you're interested in udata in https://github.com/opendatateam/udata/discussions/2721.

**Comment by syji35** [comment]
Attempt to update udata 3 -> udata 4.

Documentation doesn't seems up to date (https://udata.readthedocs.io/en/v4.0.2/) :  
- Missing explanation about SEARCH_SERVICE_API_URL and KAFKA_URI in config file  
- Change version of elastiseach in requirement ?
- link to udata-search-service project.

**Reply by maudetes** [reply]
Thank you for the feedback!

We're preparing the 4.0.3 release that should help you migrate to udata 4 more easily.
Indeed, it won't require Kafka or an external search service if not needed.

We're also clarifying the documentation on the different points you mentioned.
- SEARCH_SERVICE_API_URL and KAFKA_URI will be optional and None by default. They will be required if you want to use [udata-search-service](https://github.com/opendatateam/udata-search-service) instead of the default mongo full text search.
- elasticsearch isn't required for udata 4 to work. The elastic version required for the udata search service is 7.16.
